docs/getting-started.rst

```diff
@@ -0,0 +1,249 @@
+
+Ensure your pulumi code directory contains a ``requirements.txt`` file with at least this repo listed:
+::
+
+  git+https://github.com/thunderbird/pulumi.git
+
+You can pin your code to a specific version of this module by appending ``@branch_name`` to that. For example:
+::
+
+  git+https://github.com/thunderbird/pulumi.git@v0.0.2
+
+Pulumi will need these requirements installed. On your first run of a ``pulumi preview`` command (or some others),
+Pulumi will attempt to set up its working environment. If this fails, or you need to make adjustments later, you can
+activate Pulumi's virtual environment to perform pip changes. Assuming Pulumi's virtual environment lives at ``venv``,
+run:
+::
+
+  ./venv/bin/pip install -U -r requirements.txt
+
+You can now develop Python Pulumi code in that directory, referring to this module with imports such as these:
+::
+
+  import tb_pulumi
+
+  # ...or...
+
+  from tb_pulumi import (ec2, fargate, secrets)
+
+
+Use this module
+^^^^^^^^^^^^^^^
+
+When you issue ``pulumi`` commands (like "up" and "preview" and so on), it looks for a ``__main__.py`` file in your
+current directory and executes the code in that file. To use this module, you'll import it into that file and write up
+some code and configuration files.
+
+
+Create a config file
+""""""""""""""""""""
+
+It is assumed that a config file will exist at ``config.$STACK.yaml`` where ``$STACK`` is the currently selected Pulumi
+stack. This file must contain a mapping of names of config settings to their desired values. Currently, only one such
+setting is formally recognized. That is ``resources``.
+
+This is a mostly arbitary mapping that you will have to interpret on your own. This allows for flexibility, but we
+recommend some conventions here. Namely:
+
+* ``resources`` should be a mapping where the keys are the Pulumi type-strings for the resources they are configuring.
+  For example, if you want to build a VPC with several subnets, you might use the ``tb_pulumi.network.MultiCidrVpc``
+  class. Following this convention, that should be accompanied by a ``tb:network:MultiCidrVpc`` key in this mapping.
+* The values these keys map to should themselves be mappings. This provides a convention where more than one of each
+  pattern are configurable. The keys here should be arbitrary but unique identifiers for the resources being configured.
+  F/ex: ``backend`` or ``api``.
+* The values these keys map to should be a mapping where the keys are valid configuration options for the resources
+  being built. The full listing of these values can be found by browsing the documentation.
+
+
+Define a ThunderbirdPulumiProject
+"""""""""""""""""""""""""""""""""
+
+In your ``__main__.py`` file, start with a simple skeleton (or use ``__main__.py.example`` to start):
+::
+
+  import tb_pulumi
+  project = tb_pulumi.ThunderbirdPulumiProject()
+
+If you have followed the conventions outlined above, ``project`` is now an object with a key property, ``config``, which
```

Comment:
so it reads off config.project.yaml? Or the project is autoloaded from the config?

Reply:
Pulumi tracks the name of its project and stack internally, and then presents that context via the `pulumi` module in Python. When you construct a `ThunderbirdPulumiProject`, we pull that stack name, look for a file called `config.thatstackname.yaml`, read its contents in, and present it through the `ThunderbirdPulumiProject` as the `config` property. So it's "automatic" in the sense that the code enforces this convention. [Here is the code ref.](https://github.com/thunderbird/pulumi/blob/main/tb_pulumi/__init__.py#L58-L64)